core/src/syn/parser/stmt/define.rs

```diff
@@ -1212,6 +1216,122 @@ impl Parser<'_> {
 		Ok(res)
 	}
 
+	pub async fn parse_define_config(
+		&mut self,
+		ctx: &mut Stk,
+	) -> ParseResult<DefineConfigStatement> {
+		let (if_not_exists, overwrite) = if self.eat(t!("IF")) {
+			expected!(self, t!("NOT"));
+			expected!(self, t!("EXISTS"));
+			(true, false)
+		} else if self.eat(t!("OVERWRITE")) {
+			(false, true)
+		} else {
+			(false, false)
+		};
+
+		let next = self.next();
+		let inner = match next.kind {
+			t!("GRAPHQL") => {
+				self.pop_peek();
+				self.parse_graphql_config(ctx).await.map(ConfigInner::GraphQL)?
+			}
+			_ => unexpected!(self, next, "a type of config"),
+		};
+
+		Ok(DefineConfigStatement {
+			inner,
+			if_not_exists,
+			overwrite,
+		})
+	}
+
+	pub async fn parse_graphql_config(&mut self, ctx: &mut Stk) -> ParseResult<GraphQLConfig> {
+		use graphql::*;
+		let mut tmp_tables = Option::<TablesConfig>::None;
+		let mut tmp_fncs = Option::<FunctionsConfig>::None;
+		loop {
+			match self.peek_kind() {
+				t!("NONE") => {
+					self.pop_peek();
+					tmp_tables = Some(TablesConfig::None);
+					tmp_fncs = Some(FunctionsConfig::None);
+				}
+				t!("AUTO") => {
+					self.pop_peek();
+					tmp_tables = Some(TablesConfig::Auto);
+					tmp_fncs = Some(FunctionsConfig::Auto);
+				}
+				t!("TABLES") => {
+					self.pop_peek();
+
+					let next = self.next();
+					match next.kind {
+						t!("INCLUDE") => match ctx.run(|ctx| self.parse_value(ctx)).await? {
+							Value::Array(a) => {
+								match a
+									.into_iter()
+									.map(|v| TableConfig::try_from(v))
+									.collect::<Result<Vec<TableConfig>, _>>()
+								{
+									Ok(v) => tmp_tables = Some(TablesConfig::Include(v)),
+									Err(v) => todo!("found {v} expected "),
+								}
+							}
+							v => todo!("found {v} expected tables"),
+						},
+						t!("EXCLUDE") => match ctx.run(|ctx| self.parse_value(ctx)).await? {
+							Value::Array(a) => {
+								match a
+									.into_iter()
+									.map(|v| TableConfig::try_from(v))
```

Comment:
I see that TableConfig's can only be created from a limited number of types, so please enforce that within the syntax. If expressions and such are not allowed calling parse_value is not the right function.